youtube_dl/extractor/itunes.py

```diff
@@ -38,36 +38,35 @@ def convert_time(self, timestr):
         'Converts \'Mar 14, 2004\' strings to \'20040314\''
         if not timestr:
             return timestr
-        return datetime.datetime.strptime(timestr, '%b %d, %Y').strftime(
-            '%Y%m%d')
+        try:
+            return datetime.datetime.strptime(timestr, '%b %d, %Y').strftime(
+                '%Y%m%d')
+        except:
+            return datetime.datetime.strptime(timestr, '%d %B %Y').strftime(
+                '%Y%m%d')
```

Comment:
We need a better solution for this. Various content uses different date formats. So far I've just come across the two, but I'm sure there are more. The extractor currently fails for unknown date formats.